### PR TITLE
fix: handle boolean field properly

### PR DESCRIPTION
**Pull request by smomin**
- update field extract field for boolean data type to allow false as valid value

### CODE REVIEW

File: functions/__tests__/functions.create.test.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import algoliasearch from 'algoliasearch';
-import { firestore } from 'firebase-admin/lib/firestore';
 import * as functionsTestInit from 'firebase-functions-test';
 import mockedEnv from 'mocked-env';
 import { mocked } from 'ts-jest/utils';
 import { mockConsoleInfo } from './__mocks__/console';
-import {version} from '../src/version';
```
Comment (Haroenv) [inline functions/__tests__/functions.create.test.ts:7]: some of the changes seem unrelated

File: functions/lib/extract.js [hunk 2]
```diff
@@ -25,7 +25,8 @@ const getPayload = (snapshot) => {
         const [field, value] = processors_1.valueProcessor(firebaseField, snapshot.get(firebaseField));
         logs.debug('field', field);
         logs.debug('value', value);
-        if (value) {
+        // for fields with boolean data type, the value of false is a valid value.
+        if (value || value == false) {
```
Comment (Haroenv) [inline functions/lib/extract.js:29]:
```suggestion
        // for fields with boolean data type, the value of false is a valid value.
        if (typeof value !== 'undefined') {
```

does this also test the same?

File: functions/src/extract.ts [hunk 3]
```diff
@@ -50,7 +50,8 @@ const getPayload = (snapshot: DocumentSnapshot) => {
     logs.debug('field', field);
     logs.debug('value', value);
 
-    if (value) {
+    // for fields with boolean data type, the value of false is a valid value.
+    if (value || value == false) {
```
Comment (Haroenv) [inline functions/src/extract.ts:54]:
```suggestion
    // for fields with boolean data type, the value of false is a valid value.
    if (typeof value !== 'undefined') {
```

does this also test the same?

**Review by Haroenv — CHANGES_REQUESTED** [review]
checking specifically for undefined seems smarter than for false, since you'd need to check other falsy values too (empty string, 0)

File: functions/__tests__/functions.create.test.ts (hunk 1 above)
Comment (smomin) [reply functions/__tests__/functions.create.test.ts:7]: i had to refactor the tests to remove false positives.
